esp-wifi/CHANGELOG.md

```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Bump MSRV to 1.83 (#2615)
 - Bump smoltcp to 0.12.0 (#2849)
+- Avoid public API changes due to config options. `csi_enabled` option converted to feature (#2945)
```

Comment:
not sure the "Avoid public API changes due to config options." part is interesting for the CHANGELOG - but we have a couple of such entries

Reply:
The changelog shouldn't contain reasoning, it should just say what changed. The why is in the PR description or i the linked issue.